### PR TITLE
RFC 58: Add a new test property to control displaying test state updates

### PR DESCRIPTION
**Pull request by sefeng211**
test changes: https://github.com/web-platform-tests/wpt/pull/24170

### CODE REVIEW

File: rfcs/disable_notify_test_state.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# RFC ??: Add a new test property to not displaying test state
```
Comment (stephenmcgruer) [inline rfcs/disable_notify_test_state.md:1]:
```suggestion
# RFC 58: Add a new test property to control displaying test states
```

File: rfcs/disable_notify_test_state.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## Summary
+
+Add a new test property called `notify_test_state` to not displaying
```
Comment (stephenmcgruer) [inline rfcs/disable_notify_test_state.md:5]:
```suggestion
Add a new test property called `notify_test_state` to control displaying
```

File: rfcs/disable_notify_test_state.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+test state when tests are running.
+
```
Comment (stephenmcgruer) [inline rfcs/disable_notify_test_state.md:7]:
```suggestion
test state when tests are running.

This property will default to `true`, and can be set to `false` via the `setup()` function.
```

File: rfcs/disable_notify_test_state.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# RFC ??: Add a new test property to not displaying test state
+
+## Summary
+
+Add a new test property called `notify_test_state` to not displaying
+test state when tests are running.
```
Comment (stephenmcgruer) [inline rfcs/disable_notify_test_state.md:6]:
```suggestion
test state updates while tests are running.
```

File: rfcs/disable_notify_test_state.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+## Details
```
Comment (stephenmcgruer) [inline rfcs/disable_notify_test_state.md:8]: How does this interact with the `test_done_callbacks`? I believe those too cause `WindowTestEnvironment` to update the DOM, right?

Comment (sefeng211) [reply rfcs/disable_notify_test_state.md:8]: I don't think this impacts the `test_done_callbacks`, this PR just prevents `notify_test_state` to be invoked. We invoke the `test_done_callbacks` in `notify_result` function which is not impacted by this PR.

**Comment by jgraham** [comment]
It's kind of unclear if blocking `notify_test_state` is actually the right approach, since in theory other things could be using that (e.g. if we changed the harness to stream events back and used that to build an interactive view of the test state). It might make more sense to just make the Output handler not register a callback in case this property is set.

File: rfcs/disable_notify_test_state.md (hunk 5 above)
Comment (stephenmcgruer) [reply rfcs/disable_notify_test_state.md:8]: Yes, what I'm asking is if they could affect the tests! E.g. if a test has 10 subtests, aren't the `test_done_callbacks` invoked for each individual subtest finishing, and then each of those 10 updates causes DOM changes?

Put another way, I suspect that blocking just `notify_test_state` may not be sufficient.

**Comment by sefeng211** [comment]
@stephenmcgruer @jgraham Yeah....good point! 

How about we just not register the `show_status()` function to `test_state_callbacks` and `test_done_callbacks`? 

So this test property could be called `register_show_status_callback`?

**Comment by jgraham** [comment]
I think we should name the property for the desired effect rather than the implementation strategy. Maybe `notify_test_state` seemed OK because we would stop doing in-content notifications, but I see that it's confusing if you think of it as blocking all kinds of notifications. So we could maybe have `output_test_state: false` or `allow_harness_dom_updates: false` or something.

**Comment by sefeng211** [comment]
@jgraham sounds good, what do you think about the implementation strategy, Stephen mentioned a good point that we don't want to output anything between tests.

**Comment by jgraham** [comment]
I don't `show_results` in the output function is only called at most once when the tests are complete. We don't do interactive update of the table of results. So I think as long as `show_status` isn't called it's probably fine.

**Comment by sefeng211** [comment]
I updated the WPT patch to change the implementation from checking the flag in `notify_test_state` to not calling the `show_status` function in the callback. Detail: https://github.com/web-platform-tests/wpt/pull/24170/commits/255548294450d02c4465e62c8a434dc12faf55a5, @stephenmcgruer Please let me know what you think :)

**Comment by sefeng211** [comment]
I left the flag name as `hide_test_state` because we want the default value to be false.

File: rfcs/hide_test_state.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# RFC 58: Add a new test property to control displaying test states
+
+## Summary
+
+Add a new test property called `hide_test_state` to control displaying
+test states update while tests are running.
+
+This property will default to `true`, and can be set to `false` via the `setup()` function.
```
Comment (Ms2ger) [inline rfcs/hide_test_state.md:8]:
```suggestion
This property will default to `false`, and can be set to `true` via the `setup()` function.
```

**Comment by sefeng211** [comment]
Updated!

**Comment by sefeng211** [comment]
@Ms2ger Does it look good to you? (pinging in case it falls off your radar)

**Comment by stephenmcgruer** [comment]
Checked with jgraham that they're happy too, so that's the 1-week LGTM-no-objections time passed. Merging.

**Pull request merged**